library/src/main/java/com/novoda/downloadmanager/notifications/NotificationTag.java

```diff
@@ -0,0 +1,68 @@
+package com.novoda.downloadmanager.notifications;
+
+import android.app.Notification;
+
+import com.novoda.downloadmanager.lib.DownloadBatch;
+
+/**
+ * Build tag used for collapsing several {@link DownloadBatch} into a single
+ * {@link Notification}.
+ */
+public class NotificationTag {
+
+    private final int status;
+    private final String identifier;
+
+    public static NotificationTag create(DownloadBatch batch, String packageName) {
+        if (batch.isQueuedForWifi()) {
+            return new NotificationTag(DownloadNotifier.TYPE_WAITING, packageName);
+        } else if (batch.isRunning() && batch.shouldShowActiveItem()) {
+            return new NotificationTag(DownloadNotifier.TYPE_ACTIVE, packageName);
+        } else if (batch.isError() && !batch.isCancelled() && batch.shouldShowCompletedItem()) {
+            // Failed downloads always have unique notifications
+            return new NotificationTag(DownloadNotifier.TYPE_FAILED, String.valueOf(batch.getBatchId()));
+        } else if (batch.isCancelled() && batch.shouldShowCompletedItem()) {
+            // Cancelled downloads always have unique notifications
+            return new NotificationTag(DownloadNotifier.TYPE_CANCELLED, String.valueOf(batch.getBatchId()));
+        } else if (batch.isSuccess() && batch.shouldShowCompletedItem()) {
+            // Complete downloads always have unique notifications
+            return new NotificationTag(DownloadNotifier.TYPE_SUCCESS, String.valueOf(batch.getBatchId()));
+        } else {
+            return null;
```

Comment:
want to add the nullable annotation to the method?

Also I notice this class and this method is public, so do you expect client libraries to call it? If so it'd be good if you add javadoc to explain its use (and document the null return)